specification/dataprotection/resource-manager/Microsoft.DataProtection/stable/2023-11-01/dataprotection.json

```diff
@@ -0,0 +1,8033 @@
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataProtection/locations/{location}/fetchCrossRegionRestoreJob": {
+      "post": {
+        "tags": [
+          "DppJob"
+        ],
+        "operationId": "CrossRegionRestoreJob_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Request body for operation",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CrossRegionRestoreJobRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Details about a Cross Region Restore Job",
+            "schema": {
+              "$ref": "#/definitions/AzureBackupJobResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Cross Region Restore Job": {
+            "$ref": "./examples/CrossRegionRestore/FetchCrossRegionRestoreJob.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataProtection/locations/{location}/fetchCrossRegionRestoreJobs": {
+      "post": {
+        "tags": [
+          "DppJob"
+        ],
+        "operationId": "CrossRegionRestoreJobs_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "$filter",
```

Comment:
This API is replacement for a GET API cross region recovery scenario hence is modelled the same way along with quer parameters. Therefore we wanted to get an exception for this API to allow query parameter in POST.

Reply:
Also, we have a similar APIs for on of our other RPs where we are passing filter as query parameter in post request. We wanted to follow similar design across both RPs for simpler integration across the two.
https://github.com/Azure/azure-rest-api-specs/blob/ec238f30bd6d4a0681b691908fe00b54868467de/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2023-01-15/bms.json#L275C9-L275C9

Reply:
Is the query filter added to achieve consistency with other similar API'S?  What is the use of filter query in API contract, Isn't it used? Is this a case of using post API to retrieve data and thus needing the filters? using OData filters for the resource collection API instead of using POST is a better option here.

Reply:
Yes, POST API is being used to fetch data here. Since we need to pass a body containing details about the recovered resource to validate the request is coming from correct paired region, call cannot be a GET and had to be POST API. Filter parameters were passed in query to be consistent with other similar API in the other RP and also because we couldn't find any precedent of filters being based in request body.

Reply:
Shouldn't be an issue to pass parameters in body instead of using a filter. Is there anything blocking to pass a particular property in body?

Reply:
Nothing is blocking but as I mentioned we wanted to keep API similar to other RP for better integration across both.